### PR TITLE
[RC1] Fill Keystone testing

### PR DESCRIPTION
**Pull request by collivier**
It describes Keystone testing in RC1 4.3.4.1.

Closes: #830

Signed-off-by: Cédric Ollivier <cedric.ollivier@orange.com>

### CODE REVIEW

**Comment by collivier** [comment]
@rabi-abdel @rabiabdel labeler is failing here, right?

File: doc/ref_cert/lfn/chapters/chapter04.md [hunk 1]
```diff
@@ -67,21 +67,45 @@ Note that each requirement may have one or more test cases associated with it.
...
+- .\*identity.v2
+
+Keystone API is also covered by [Rally](https://opendev.org/openstack/rally).
+
+Here are the mainline tasks integrated in
+[Functest Smoke CNTT](https://git.opnfv.org/functest/tree/docker/smoke-cntt/testcases.yaml):
+- Authenticate.keystone
+- KeystoneBasic.add_and_remove_user_role
+- KeystoneBasic.create_add_and_list_user_roles
+- KeystoneBasic.create_and_list_tenants
```
Comment (mf4716) [inline doc/ref_cert/lfn/chapters/chapter04.md]: @collivier - can we create a table which traces these tasks to actual requirements?  The ask is to create TC Traceability to RM/RA requirements, so we need to list out TCs and their respective requirements, which also shows we're either testin exactly what has been asked, or if testing is going above and beyond what is needed.  A table mapping requirements and TCs will be great visual aid.

Comment (deebhatia) [reply doc/ref_cert/lfn/chapters/chapter04.md]: @mf4716 You mean something like [this](https://github.com/cntt-n/CNTT/issues/830#issuecomment-568427423) ?

Comment (collivier) [reply doc/ref_cert/lfn/chapters/chapter04.md]: @mf4716 @deebhatia 
You may be not aware about the OpenStack gates and how tempest is designed.
Here it's straightforward: it executes all tempest testcases except the optional capabilities in RA1 Chapter 5. 

It answers the simple requirements asked by RA1 Chapter 5: test Keystone ...
I can map the optional services and the tests not exected if it helps.

As answered in https://github.com/cntt-n/CNTT/issues/830#issuecomment-568427423, I don't see any technical reasons to duplicate the upstream work and then to list the ~2000 tempest tests already executed in Functest CNTT. It costs lots of time for a duplication in the best cases.
See opnfv/functest-smoke-cntt:hunter in https://build.opnfv.org/ci/view/functest/job/functest-hunter-daily/373/

**Comment by collivier** [comment]
@mf4716 The only requirement is to cover Keystone as described in RA1 Chapter 5. I don't see your technical point.

File: doc/ref_cert/lfn/chapters/chapter04.md (hunk 1 above)
Comment (mf4716) [reply doc/ref_cert/lfn/chapters/chapter04.md]: > @mf4716 You mean something like [this](https://github.com/cntt-n/CNTT/issues/830#issuecomment-568427423) ?

Yes @deebhatia - exactly like that.  For now, I'll approve the initial content in this PR, but we need to iterate on this for when we do our RC-1 alpha field trial.  For Snezka, its a start to have the requirements / cases listed, but we need traceability for certification and to normalize results.  Without it, we should not expect to receive a Certification Badge.  Approving.

**Comment by collivier** [comment]
For my understanding, it's already good here.
RA1 Chapter defines all Service API and capabilities and this PR introduces the mapping to Chapter5.

The possible misunderstanding comes from the fact that we do consider all tempest tests except the ones covering optional capabilities. It's a "not" logic which differs from the basic one listing all tests one by one which quickly fails in case of OpenStack compliance.

**Comment by collivier** [comment]
@deebhatia @rabi-abdel @rabiabdel : Could you please approve this change? All other pull-requests from this chains are well rated and (and then ready for merge)

File: doc/ref_cert/lfn/chapters/chapter04.md [hunk 2]
```diff
@@ -75,11 +75,38 @@ Note: It will only target the functionality that are exposed by standard OpenSta
...
-|----|------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------|
-| `cert.test.vim.01` | API | Show API endpoint catalog. It *must* have endpoint for each Core service. | `req.int.api.01` |
+Keystone API is covered in the OpenStack Gates via
+[Tempest](https://opendev.org/openstack/tempest) and
+[keystone-tempest-plugin](https://opendev.org/openstack/keystone-tempest-plugin)
+as integrated in
+[Functest Smoke CNTT](https://git.opnfv.org/functest/tree/docker/smoke-cntt/testcases.yaml).
+
+According to
+[RA1 Core OpenStack Services APIs](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/openstack/chapters/chapter05.md)
```
Comment (deebhatia) [inline doc/ref_cert/lfn/chapters/chapter04.md]: Use relative links. This will redirect wrongly when compiled with jekyll on `https://cntt-n.github.io/`

**Pull request merged**